helium-lib/src/transaction.rs

```diff
@@ -0,0 +1,157 @@
+use crate::{
+    client::SolanaClient,
+    error::Error,
+    priority_fee::auto_compute_limit_and_price,
+    solana_client::{
+        nonblocking::tpu_client::TpuClient,
+        send_and_confirm_transactions_in_parallel::{
+            send_and_confirm_transactions_in_parallel, SendAndConfirmConfig,
+        },
+        tpu_client::TpuClientConfig,
+    },
+    solana_sdk::{
+        commitment_config::CommitmentConfig, compute_budget::ComputeBudgetInstruction,
+        instruction::Instruction, message::Message, signature::Keypair, signer::Signer,
+        transaction::Transaction,
+    },
+};
+
+const MAX_TRANSACTION_SIZE: usize = 1232; // Maximum transaction size in bytes
+
+pub fn replace_or_insert_instruction(
+    instructions: &mut Vec<Instruction>,
+    new_instruction: Instruction,
+    insert_pos: usize,
+) {
+    if let Some(pos) = instructions
+        .iter()
+        .position(|ix| ix.program_id == solana_sdk::compute_budget::id())
+    {
+        instructions[pos + insert_pos] = new_instruction;
+    } else {
+        instructions.insert(insert_pos, new_instruction);
+    }
+}
+
+// Returns packed txs with the indices in instructions that were used in that tx.
+pub fn pack_instructions(
```

Comment:
I see the usecase for it, but I'm still not sure where we actually need this at all for the on chain oui work. Those are all known fixed instructions going into a transaction at a time. I'd like to lift the user of this library to a higher level than "instructions"

Reply:
Fair, I think originally I was packing multiple ixs into n txs and was using this. I dumbed down the ixs for ouis so will remove this